src/__tests__/VoucherRedemption.tsx

```diff
@@ -0,0 +1,263 @@
+      item_id: 4,
+      seller_gift_card_id:
+        'crlmuwcorvfwrhnzdfkdjkjhwwitqqwnhhelyahrctehykfustgvrbalipvwiaus',
+      recipient_id: 1,
+      single_use: false,
+      amount: 300,
+    },
+  };
+
+  return {
+    ...jest.requireActual('../utilities/api/interactionManager'),
+    getSeller: jest.fn(() => ({
+      data: shunfaBakeryResponse,
+    })),
+    getVoucher: jest.fn(() => ({
+      data: voucherResponse,
+    })),
+    updateVoucher: jest.fn(() => ({
+      data: voucherUpdateResponse,
+    })),
+  };
+});
+
+beforeAll(() => {
+  axios.defaults.adapter = require('axios/lib/adapters/http');
+});
+
+afterEach(() => cleanup());
+
+const renderVoucher = (providerValues) => (
+  <VoucherProvider>
+    <VoucherPage />
+  </VoucherProvider>
+);
+
+describe('Voucher Page', () => {
+  let providerValues = {};
+  let VoucherProps = {};
+  let VoucherDispatcher = jest.fn();
+  beforeEach(() => {
+    VoucherProps = VoucherTypes.defaultState;
+    providerValues = {
+      VoucherProps,
+      VoucherDispatcher,
+    };
+  });
+
+  test('should display VoucherPage with succesful GET /voucher request', async () => {
+    const { findByText } = renderIntegration(
+      '/voucher/123asdf',
+      renderVoucher(providerValues)
+    );
+
+    const HeaderLabel = await findByText(`Welcome to`, { exact: false });
+    const Header = await findByText(`Shunfa Bakery`, { exact: false });
+    const CardHeader = await findByText('Your available balance');
+    const Amount = await findByText('$25.00');
+    const CodeLabel = await findByText('Voucher Code', { exact: false });
+    const Code = await findByText('123asdf', { exact: false });
+    const NextButton = await findByText(
+      'Click to begin redeeming your voucher'
+    );
+
+    expect(HeaderLabel).toBeInTheDocument();
+    expect(Header).toBeInTheDocument();
+    expect(CardHeader).toBeInTheDocument();
+    expect(Amount).toBeInTheDocument();
+    expect(CodeLabel).toBeInTheDocument();
+    expect(Code).toBeInTheDocument();
+    expect(NextButton).toBeInTheDocument();
+  });
+
+  test('should display VoucherPage Amount prompt when clicking "Click to being to being reddeming your voucher"', async () => {
+    const { findByText, findAllByText } = renderIntegration(
+      '/voucher/123asdf',
+      renderVoucher(providerValues)
+    );
+
+    const NextButton = await findByText(
+      'Click to begin redeeming your voucher'
+    );
+
+    fireEvent.click(NextButton);
+
+    const Header = await findByText(`Shunfa Bakery`, { exact: false });
+    const CurrentBalanceHeader = await findByText('Current balance');
+    const Amount = await findAllByText('25.00', { exact: false });
+    const Prompt = await findByText('How much are you spending today?');
+    const Code = await findByText('123asdf', { exact: false });
+    const CodeLabel = await findByText('Voucher Code', { exact: false });
+    const Next = await findByText('Next', { exact: false });
```

Comment:
Some of these checks look like they can be abstracted into a method and shared

Reply:
It is possible. For this particular test we could just because there is a constant theme of checking for the voucher code, its label and the store header. I could add that in the final commit